framework/src/main/java/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java

```diff
@@ -149,304 +162,209 @@ public boolean hasDependentAnnotations() {
-            if (javaExpression != null) {
+            if (javaExpressionAnno != null) {
                 elements.add(method.getName());
             }
         }
         return elements;
     }
 
-    ///
-    /// Viewpoint adaptation
-    ///
+    /**
+     * Returns the elements of the annotation that are Java expressions.
+     *
+     * @param am AnnotationMirror
+     * @return the elements of the annotation that are Java expressions
+     */
+    private List<String> getListOfExpressionElements(AnnotationMirror am) {
+        for (Class<? extends Annotation> clazz : annoToElements.keySet()) {
+            if (factory.areSameByClass(am, clazz)) {
+                return annoToElements.get(clazz);
+            }
+        }
+        return Collections.emptyList();
+    }
+
+    /**
+     * Creates a TreeAnnotator that viewpoint-adapts dependent type annotations.
+     *
+     * @return a new TreeAnnotator that viewpoint-adapts dependent type annotations
+     */
+    public TreeAnnotator createDependentTypesTreeAnnotator() {
+        assert hasDependentAnnotations();
+        return new DependentTypesTreeAnnotator(factory, this);
+    }
 
     /**
      * Viewpoint-adapts the dependent type annotations on the bounds to the use of the type.
      *
      * @param classDecl class or interface declaration whose type variables should be viewpoint
      *     adapted
-     * @param bounds annotated types of the bounds of the type variables; side-effected by this
-     *     method
+     * @param bounds annotated types of the bounds of the type variables; its elements are
+     *     side-effected by this method (but the list itself is not side-effected)
      */
-    public void viewpointAdaptTypeVariableBounds(
+    public void atTypeVariableBounds(
             TypeElement classDecl, List<AnnotatedTypeParameterBounds> bounds) {
         if (!hasDependentAnnotations()) {
             return;
         }
-        JavaExpression r = JavaExpression.getImplicitReceiver(classDecl);
-        JavaExpressionContext context = new JavaExpressionContext(r, factory.getChecker());
+
+        StringToJavaExpression stringToJavaExpr =
+                stringExpr ->
+                        StringToJavaExpression.atTypeDecl(
+                                stringExpr, classDecl, factory.getChecker());
         for (AnnotatedTypeParameterBounds bound : bounds) {
-            viewpointAdaptToContext(context, bound.getUpperBound());
-            viewpointAdaptToContext(context, bound.getLowerBound());
+            convertAnnotatedTypeMirror(stringToJavaExpr, bound.getUpperBound());
+            convertAnnotatedTypeMirror(stringToJavaExpr, bound.getLowerBound());
         }
     }
 
     /**
-     * Viewpoint-adapts the dependent type annotations in the methodDeclType based on the
+     * Viewpoint-adapts the dependent type annotations in the methodType based on the
```

Comment:
The documentation uses the terms "based on", "for", and "to".  Please make them consistent.  I like "to" myself, as in "viewpoint-adapt T to its declaration" and that is already used in parts of the documentation, but I'm open to other possibilities if they are better.